### PR TITLE
en_invadepoh OK

### PR DESCRIPTION
**Pull request by kyleburnette**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Build is OK and all functions are decomped!

Time for cleanup and documentation. Definitely gonna need help on that.

### CODE REVIEW

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:3050]:
```suggestion
    if ((globalCtx->gameplayFrames % 64) < 14) {
```

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:3107]:
```suggestion
    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);

    if (this->actionTimer > 0) {
        this->actionTimer--;
    } else {
        func_80B49904(this);
    }
```

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:3423]:
```suggestion
    if ((globalCtx->gameplayFrames % 256) == 0) {
```

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:3519]: Ehh, what?
Did you meant this?
```suggestion
        this->actor.flags &= ~0x9;
```

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:3672]:
```suggestion
    if (gSaveContext.weekEventReg[0x16] & 1) {
```

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:1]: Could you add the actor description?

**Review by AngheloAlf — COMMENTED** [review]
This actor is huge! D:
Very good work.
I made a lot of comments, but you could ignore most of them if you want.
Btw, I really liked the `CLOCK_TIME` macro.

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (engineer124) [inline src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:2349]:
```suggestion
        Item_DropCollectible(globalCtx, &this->actor.world.pos, ITEM00_ARROWS_30);
```
That makes sense right?

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (engineer124) [inline src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:3284]:
```suggestion
        diff = BINANG_SUB(this->actor.yawTowardsPlayer, this->actor.shape.rot.y);
```
While this would fail now, this macro is being introduced in https://github.com/zeldaret/mm/pull/202/files

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (engineer124) [inline src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:3910]:
```suggestion
    if (gSaveContext.time > 0xD800) {
```

**Review by engineer124 — COMMENTED** [review]
Again, well done on this massive actor! Just a quick skim of things I noticed, I'll wait for a clean-up/documentation for a thorough review :)

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (kyleburnette) [reply src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:1102]: That doesn't match, but this does (not sure if it's better but I kinda like it) `    if (this->actor.floorHeight > BGCHECK_Y_MIN + 1)`

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (kyleburnette) [reply src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:2349]: The aliens drop those when you kill them so I think it makes sense

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (engineer124) [reply src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:1102]: Nice! I like it too

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (kyleburnette) [reply src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:991]: This does match. I personally like the increment outside of the condition, but I don't feel too strongly either way

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (Kenix3) [reply src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:3264]: #202 is merged, and you have a conflict. Please use the macro when you fix the conflict. Thanks!

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:660]: From what I saw, those `Actor*` variables seem to be `EnInvadepoh`, so you could probably change them.

File: tools/decomp-permuter/permuter_settings.toml
Comment (kyleburnette) [reply tools/decomp-permuter/permuter_settings.toml:12]: Oh nice, I didn't see it there actually. I'll remove this one.

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (kyleburnette) [reply src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:1060]: For this one I can move sp40 up, but not sp60

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (kyleburnette) [reply src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:1485]: This gives a 1 line issue (operands of an addu are swapped) but I agree, there's definitely a better match to be had here somehow

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (engineer124) [inline src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:4670]:
```suggestion
    gDPSetPrimColor(POLY_XLU_DISP++, 0xFF, 0x80, 255, 255, 0, 180);
    gDPSetEnvColor(POLY_XLU_DISP++, 255, 50, 0, 0);
```

File: src/overlays/actors/ovl_En_Ma4/z_en_ma4.c
Comment (engineer124) [inline src/overlays/actors/ovl_En_Ma4/z_en_ma4.c:666]: In OoT, it's been consistent to have underscores between the `unk` and the number. In MM, I haven't been too picky about it but I should bring it up that it's generally been inconsistent in MM atm

**Review by engineer124 — COMMENTED** [review]
Time for my in-depth review! I'm sorry for the large number of comments but don't be afraid. I swear it feels like over half of them are just simply clean dec to non-clean hex angles, and so many more comments are just repeats of previous comments (there's a lot of repeated code snippets in this actor). Overall, great work again!

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (hensldm) [reply src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:3869]: Could you also change `0.0001f` to `SQ(0.01f)` since the function is returned distance squared?

File: src/overlays/actors/ovl_En_Ma4/z_en_ma4.c
Comment (Kenix3) [reply src/overlays/actors/ovl_En_Ma4/z_en_ma4.c:666]: MM is supposed to not have the underscore.

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.h
Comment (engineer124) [reply src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.h:22]: I partially take this comment back as there are a few instances of snake-case for struct names, so I'll leave it up to you

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (kyleburnette) [reply src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:1495]: Unfortunately not :(

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (kyleburnette) [reply src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:970]: For some reason a clamp does not work here

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (kyleburnette) [reply src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:2726]: This actually doesn't match it seems

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (kyleburnette) [reply src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:3538]: 99.99% sure it's milk you get from defeating aliens so I'm gonna roll with that

File: tools/timeconv.py
Comment (hensldm) [reply tools/timeconv.py:20]: Since only the HH,MM clock time macro was added, we should probably include Tharo's version of this timeconv script that does not include seconds.

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (kyleburnette) [reply src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:4447]: For some reason this seems to be needed

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c
Comment (kyleburnette) [reply src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:1144]: I think I'll leave them for now until documentation gets far enough along to know what versions of invadepoh are "doing nothing" in these cases

File: include/z64item.h
Comment (engineer124) [inline include/z64item.h:93]:
```suggestion
    /* 0x60 */ GI_MILK_BOTTLE = 0x60,
```
Really nice work addressing all the comments. Hopefully it wasn't too bad. Last comment from me, I double checked OoT and this is the name they use

**Review by engineer124 — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**
